File: maximum_meetings_one_room.py
from typing import List

class Meet:
    def __init__(self,a,b,c):
        self.start=a
        self.end=b
        self.index=c

class Solution:
    def maxMeetings(self, N : int, S : List[int], F : List[int]) -> List[int]:
        # code here
        times=[]
        
        for i in range(N):
            times.append(Meet(S[i],F[i],i+1))
        
        times.sort(key=lambda x: (x.end,x.index))
        
        ans=[]
        lastend=-1
        
        for i in range(N):
            if times[i].start>lastend:
                ans.append(times[i].index)
                lastend=times[i].end
        ans.sort()
        
        return ans
        
        



#{ 
 # Driver Code Starts
class IntArray:
    def __init__(self) -> None:
        pass
    def Input(self,n):
        arr=[int(i) for i in input().strip().split()]#array input
        return arr
    def Print(self,arr):
        for i in arr:
            print(i,end=" ")
        print()


if __name__=="__main__":
    t = int(input())
    for _ in range(t):
        
        N = int(input())
        
        
        S=IntArray().Input(N)
        
        
        F=IntArray().Input(N)
        
        obj = Solution()
        res = obj.maxMeetings(N, S, F)
        
        IntArray().Print(res)
        

# } Driver Code Ends

File: maximum_meetings_one_room.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class meet{
    public:
    int start,end,index;
    meet(int a,int b,int c){
        start=a;
        end=b;
        index=c;
    }
};

class Solution{
public:
    vector<int> maxMeetings(int N,vector<int> &S,vector<int> &F){
        vector<meet> times;
        for(int i=0;i<N;i++){
            times.emplace_back(meet(S[i],F[i],i+1));
        }
        
        sort(begin(times),end(times), [](meet &a,meet  &b){
           if(a.end==b.end) return a.index<b.index;
           
           return a.end<b.end;
        });
        
        vector<int> ans;
        int lastend=-1;
        for(int i=0;i<N;i++){
            if(times[i].start>lastend)
            {
                ans.push_back(times[i].index);
                lastend=times[i].end;
            }
        }
        sort(begin(ans),end(ans));
        
        return ans;
    }
};


//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        vector<int> S(n),F(n);
        for(int i=0;i<n;i++){
            cin>>S[i];
        }
        for(int i=0;i<n;i++){
            cin>>F[i];
        }
        Solution ob;
        vector<int> ans=ob.maxMeetings(n,S,F);
        for(int i=0;i<ans.size();i++){
            cout<<ans[i];
            if(i!=ans.size()-1){
                cout<<" ";
            }
        }
        cout<<endl;
    }
    return 0;
}
// } Driver Code Ends